src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1017,14 +1022,11 @@ trait Contexts { self: Analyzer =>
         )
       )
       def lookupInPrefix(name: Name)    = {
-        val sym = pre.member(name).filter(qualifies)
-        def isNonPackageNoModuleClass(sym: Symbol) =
-          sym.isClass && !sym.isModuleClass && !sym.isPackageClass
-        if (!sym.exists && unit.isJava && isNonPackageNoModuleClass(pre.typeSymbol)) {
-          // TODO factor out duplication with Typer::inCompanionForJavaStatic
-          val pre1 = companionSymbolOf(pre.typeSymbol, this).typeOfThis
-          pre1.member(name).filter(qualifies).andAlso(_ => pre = pre1)
-        } else sym
+        val searchCompanions = isPlainClassInJavaUnit(pre.typeSymbol)
+        val sym = pre.findMember(name, excludedFlags = Flags.BridgeFlags, requiredFlags = 0, stableOnly = false, searchCompanions).filter(qualifies)
+        if (searchCompanions && sym.owner.isModuleClass)
+          pre = sym.owner.typeOfThis
```

Comment:
`sym` is a member, and `sym.owner` is a module class. I think when completing a lazy type of a class, its members' types are not completed.